var/spack/repos/builtin/packages/hipsycl/package.py

```diff
@@ -132,20 +134,26 @@ def filter_config_file(self):
         #    the libc++.so and libc++abi.so dyn linked to the sycl
         #    ptx backend
         rpaths = set()
-        so_paths = filesystem.find(self.spec["llvm"].prefix, "libc++.so")
+        so_paths = filesystem.find(
+                        self.spec["llvm"].prefix,
+                        "libc++.{0}".format(dso_suffix)
+                   )
```

Comment:
Here you can use `filesystem.find_libraries` which takes care of suffixes automatically.

Reply:
had to read the `llnl/util/find_libraries.py` source, but here it is.